docs_src/dependencies/tutorial013.py

```diff
@@ -0,0 +1,38 @@
+from contextlib import contextmanager
+
+from fastapi import Depends, FastAPI
+from pydantic import BaseSettings
+
+
+class Database:
+
+    def __init__(self, hostname: str) -> None:
+        ...
+
+    @contextmanager
+    def connect(self):
+        yield
+
+
+
+class AppConfig(BaseSettings):
+    database_hostname: str = "localhost"
+
+
+def Config() -> AppConfig:
+    return AppConfig()
+
+
+def DBConnection(config: AppConfig = Depends(Config)) -> Database:
```

Comment:
```suggestion
@contextlib.asynccontextmanager
def DBConnection(config: AppConfig = Depends(Config)) -> Generator[Database, None, None]:
```

Pushing the responsibility of managing the generator down to the user solves a number of weird edge cases around Exceptions/StopIteration and Traceback preservation

Reply:
makes sense, thanks!